### PR TITLE
chore: improve eslint rules

### PR DESCRIPTION
**Pull request by KelvinOm**
## Description
Make eslint rules stricter for packages

#### Type of change
- Chore (housekeeping or task changes that don't impact user perception)

- [x] Manual
- [x] Jest
- [ ] Cypress

#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### CODE REVIEW

File: app/client/packages/design-system/headless/src/components/Switch/Switch.tsx [hunk 1]
```diff
@@ -24,7 +24,7 @@ export interface SwitchProps
 
 export type SwitchRef = FocusableRef<HTMLLabelElement>;
 
-export const Switch = forwardRef((props: SwitchProps, ref: SwitchRef) => {
+const _Switch = (props: SwitchProps, ref: SwitchRef) => {
```
Comment (KelvinOm) [inline app/client/packages/design-system/headless/src/components/Switch/Switch.tsx:27]: This is fix for `react/display-name` rule. cc @jsartisan

**Review by ichik — APPROVED** [review]

File: app/client/.eslintrc.js [hunk 2]
```diff
@@ -13,6 +13,9 @@ const baseNoRestrictedImports =
 const eslintConfig = {
   extends: ["./.eslintrc.base.json"],
   rules: {
+    "@typescript-eslint/no-explicit-any": "off",
```
Comment (jsartisan) [inline app/client/.eslintrc.js:16]: won't this encourage people to use any as they like? any is mostly meant for exceptions or where it does not matter if you use any or specific type

File: app/client/packages/design-system/headless/src/components/Switch/Switch.tsx (hunk 1 above)
Comment (jsartisan) [reply app/client/packages/design-system/headless/src/components/Switch/Switch.tsx:27]: Oh cool. I saw this pattern somewhere else to ( either mantine or spectrum )

File: app/client/.eslintrc.js (hunk 2 above)
Comment (KelvinOm) [reply app/client/.eslintrc.js:16]: Exactly and it's the reason that I moved this rule from [.eslintrc.base.json](https://github.com/appsmithorg/appsmith/pull/26056/files#diff-838aa936dc1d06f2a2b8245a288cb8224e42a076559dec275313db059cc0f43aR31), which also extended to packages. The client now has 2899 errors according to this rule. I can't fix them right now. We need a comprehensive solution to fix this situation. I'm thinking about it.

**Review by jsartisan — APPROVED** [review]

**Review by riodeuno — APPROVED** [review]

**Pull request merged**